{
  while (serverSocket != null && !serverStop) {
    try {
      serverSocket.accept();
    }
 catch (    SocketException se) {
      LOG.debug("SocketException during tearDown. Can be safely ignored");
    }
catch (    IOException e) {
      LOG.error("Unexpected exception while accepting connection request");
    }
  }
}
